### PR TITLE
Add test for complex join driven by feature chaining configuration

### PR DESCRIPTION
**Pull request by ridethepenguin**

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java [hunk 1]
```diff
@@ -336,30 +363,179 @@ public void testXrefJoinHandler() {
...
+		renameCell.setTarget(getNestedUnitLocalIdTargetProperty());
+		alignment.addCell(renameCell);
+
+		DefaultCell renameClassCell = new DefaultCell();
+		renameClassCell.setTransformationIdentifier(RenameFunction.ID);
+		renameClassCell.setSource(getObservationClassSourceProperty());
+		renameClassCell.setTarget(getNestedObservationClassTargetProperty());
+		alignment.addCell(renameClassCell);
+
+//		processJoinAlignment(alignment, chainingConf);
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java]: Removed the comment here to make the test run successfully. Hope that was right.

Comment (ridethepenguin) [inline io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java]: Wow, thanks, I wonder how that escaped me... must have been sleeping or something

Comment (ridethepenguin) [inline io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java]: I also forgot to comment out (or remove) the two `logMapping` calls in the following lines... could you do it for me, please?

Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.appschema.test/src/eu/esdihumboldt/hale/io/appschema/writer/AppSchemaMappingTest.java]: Done :smile:

**Pull request merged**